docs/JAX_basic_text_classification.md

```diff
@@ -0,0 +1,455 @@
+
+```{code-cell} ipython3
+import numpy as np
+
+import tiktoken
+import grain.python as grain
+
+
+tokenizer = tiktoken.get_encoding("gpt2")
+
+
+class TextPreprocessing(grain.MapTransform):
+    def __init__(self, tokenizer, max_length: int = 256):
+        self.tokenizer = tokenizer
+        self.max_length = max_length
+
+    def map(self, data):
+        text = data["text"]
+        encoded = self.tokenizer.encode(text)
+        # Cut to max length
+        encoded = encoded[:self.max_length]
+        # Pad with zeros if needed
+        encoded = np.array(encoded + [0] * (self.max_length - len(encoded)))
+        return {
+            "text": encoded,
+            "label": data["label"],
+        }
+```
+
+```{code-cell} ipython3
+seed = 12
+train_batch_size = 128
+test_batch_size = 2 * train_batch_size
+
+# max length of tokenized text
+max_length = 500
+vocab_size = tokenizer.n_vocab
+
+
+train_sampler = grain.IndexSampler(
+    len(train_dataset),
+    shuffle=True,
+    seed=seed,
+    shard_options=grain.NoSharding(),  # No sharding since this is a single-device setup
+    num_epochs=1,                      # Iterate over the dataset for one epoch
+)
+
+test_sampler = grain.IndexSampler(
+    len(test_dataset),
+    shuffle=False,
+    seed=seed,
+    shard_options=grain.NoSharding(),  # No sharding since this is a single-device setup
+    num_epochs=1,                      # Iterate over the dataset for one epoch
+)
+
+
+train_loader = grain.DataLoader(
+    data_source=train_dataset,
+    sampler=train_sampler,                 # Sampler to determine how to access the data
+    worker_count=4,                        # Number of child processes launched to parallelize the transformations among
+    worker_buffer_size=2,                  # Count of output batches to produce in advance per worker
+    operations=[
+        TextPreprocessing(tokenizer, max_length=max_length),
+        grain.Batch(train_batch_size, drop_remainder=True),
+    ]
+)
+
+test_loader = grain.DataLoader(
+    data_source=test_dataset,
+    sampler=test_sampler,                  # Sampler to determine how to access the data
+    worker_count=4,                        # Number of child processes launched to parallelize the transformations among
+    worker_buffer_size=2,                  # Count of output batches to produce in advance per worker
+    operations=[
+        TextPreprocessing(tokenizer, max_length=max_length),
+        grain.Batch(test_batch_size),
+    ]
+)
+```
+
+```{code-cell} ipython3
+train_batch = next(iter(train_loader))
+```
+
+```{code-cell} ipython3
+print("Train encoded text batch info:", type(train_batch["text"]), train_batch["text"].shape, train_batch["text"].dtype)
+print("Train labels batch info:", type(train_batch["label"]), train_batch["label"].shape, train_batch["label"].dtype)
+```
+
+```{code-cell} ipython3
+print("Batch data:", train_batch["text"][:2, :12], train_batch["label"][:2])
```

Comment:
Can you leave a tiny note about why we are printing this, and what we should expect to see?